tests/administration/realms.test.js

```diff
@@ -1,7 +1,54 @@
+'use strict';
 const common = require('../common'),
       expect = require('chai').expect;
 
 describe('Realms', () => {
   before(() => common.login(browser, 'admin', 'admin', 'local'));
+  after(() => common.logout());
   it('should navigate to Realms', () => common.clickSidebarTab(browser, 'Realms'));
+
+  it('should create a valid realm', () => {
+  	return common.waitForOnClick('//*[@id="add_realm"]')
+  		.then(() => browser.click('//*[@id="add_realm"]'))
+  		.then(() => browser.waitUntil( browser.getAttribute('//*[@id="realm"]','onchange')
+  			.then((onc) => onc !== null )
+  		))
+  		.then(() => browser.setValue('//*[@id="realm"]', 'test'))
+  		.then(() => browser.waitUntil( browser.getAttribute('//*[@id="fqdn"]','onchange')
+  			.then((onc) => onc !== null )
+  		))
+  		.then(() => browser.setValue('//*[@id="fqdn"]','hiveio.local'))
+  		.then(() => browser.waitForExist('//*[@id="add_realm_form"]/button[1]'))
+  		.then(() => browser.click('//*[@id="add_realm_form"]/button[1]'))
+  		.then(() => browser.waitForExist('//*[@id="realm-message"]'))
+  		.then(() => browser.click('//*[@id="add_realm_form"]/button[1]'))
+  		.then(() => browser.waitForExist('//*[@id="page-wrapper"]//td[1 and text()="TEST"]'));
+  });
+
+  it('should return to page once add realm is cancelled', () => {
```

Comment:
With the current log in/log off - this test will never really be "after realm is cancelled" as the system will log off, and the log in again...